code/modules/events/apc_overload.dm

```diff
@@ -0,0 +1,63 @@
+#define APC_BREAK_PROBABILITY 25 // the probability that a given APC will be broken
+
+/datum/event/apc_overload
+	var/const/announce_after_mc_ticks     = 5
+	var/const/delayed                     = FALSE
+	var/const/event_max_duration_mc_ticks = announce_after_mc_ticks * 2
+	var/const/event_min_duration_mc_ticks = announce_after_mc_ticks
+
+	announceWhen = announce_after_mc_ticks
+
+/datum/event/apc_overload/setup()
+	endWhen = rand(event_min_duration_mc_ticks, event_max_duration_mc_ticks)
+
+/datum/event/apc_overload/start()
+	apc_overload_failure(announce=delayed)
+	var/sound/S = sound('sound/effects/powerloss.ogg')
+	for(var/mob/living/M in GLOB.player_list)
+		var/turf/T = get_turf(M)
+		if(!M.client || !is_station_level(T.z))
+			continue
+		SEND_SOUND(M, S)
+
+/datum/event/apc_overload/announce()
+	GLOB.event_announcement.Announce("Abnormal activity detected in [station_name()]'s powernet. As a precautionary measure, the station's power will be shut off for an indeterminate duration.", "Critical Power Failure", new_sound = 'sound/AI/poweroff.ogg')
+
+/datum/event/apc_overload/end()
+	return TRUE
+
+/proc/apc_overload_failure(announce=TRUE)
+	var/list/skipped_areas_apc = list(
+		/area/engine/engineering,
+		/area/turret_protected/ai)
+
+	if(announce)
+		GLOB.event_announcement.Announce("Abnormal activity detected in [station_name()]'s powernet. As a precautionary measure, the station's power will be shut off for an indeterminate duration.", "Critical Power Failure", new_sound = 'sound/AI/poweroff.ogg')
```

Comment:
Use:
Overload detected in [station_name()]'s powernet. Engineering, please replace destroyed APCs.